notebook/example.livemd

```diff
@@ -0,0 +1,143 @@
+```
+
+Go to http://localhost:5001
+
+<!-- livebook:{"attrs":{"action":":index","code":"defmodule Server.HiLive do\n  use Phoenix.LiveView, layout: {__MODULE__, :layout}\n  use LiveViewNative.LiveView\n\n  def layout(assigns) do\n    ~H\"\"\"\n      <%= @inner_content %>\n    \"\"\"\n  end\n\n  @impl true\n  def render(%{platform_id: :swiftui} = assigns) do\n    ~Z\"\"\"\n    <Text modifiers={@native |> foreground_style(primary: {:color, :mint})}>\n      Hi from LiveView Native!\n    </Text>\n    \"\"\"swiftui\n  end\n\n  def render(assigns) do\n    ~H\"\"\"\n    <div>Hi from Live</div>\n    \"\"\"\n  end\nend\n","path":"/hi"},"chunks":[[0,488],[490,2],[494,44]],"kind":"Elixir.KinoLiveViewNative","livebook_object":"smart_cell"} -->
+
+```elixir
+defmodule Server.HiLive do
+  use Phoenix.LiveView, layout: {__MODULE__, :layout}
+  use LiveViewNative.LiveView
+
+  def layout(assigns) do
+    ~H"""
+      <%= @inner_content %>
+    """
+  end
+
+  @impl true
+  def render(%{platform_id: :swiftui} = assigns) do
+    ~Z"""
+    <Text modifiers={@native |> foreground_style(primary: {:color, :mint})}>
+      Hi from LiveView Native!
+    </Text>
+    """swiftui
+  end
+
+  def render(assigns) do
+    ~H"""
+    <div>Hi from Live</div>
+    """
+  end
+end
+|> KinoLiveViewNative.register("/hi", ":index")
+```
+
+http://localhost:5001/hi
+
+<!-- livebook:{"attrs":{"action":":index","code":"defmodule Server.HelloNameLive do\n  use Phoenix.LiveView, layout: {__MODULE__, :layout}\n  use LiveViewNative.LiveView\n\n  def layout(assigns) do\n    ~H\"\"\"\n      <%= @inner_content %>\n    \"\"\"\n  end\n\n  @impl true\n  def mount(%{\"name\" => name}, _session, socket) do\n    {:ok, \n    socket\n    |> assign(name: name)}\n  end\n\n  @impl true\n  def render(%{platform_id: :swiftui} = assigns) do\n    ~Z\"\"\"\n    <Text modifiers={@native |> foreground_style(primary: {:color, :mint})}>\n      Hello <%= @name %>\n    </Text>\n    \"\"\"swiftui\n  end\n\n  def render(assigns) do\n    ~H\"\"\"\n    <div>Hello <%= @name %></div>\n    \"\"\"\n  end\nend\n","path":"/hello/:name"},"chunks":[[0,616],[618,2],[622,53]],"kind":"Elixir.KinoLiveViewNative","livebook_object":"smart_cell"} -->
+
+```elixir
+defmodule Server.HelloNameLive do
+  use Phoenix.LiveView, layout: {__MODULE__, :layout}
+  use LiveViewNative.LiveView
+
+  def layout(assigns) do
+    ~H"""
+      <%= @inner_content %>
+    """
+  end
+
+  @impl true
+  def mount(%{"name" => name}, _session, socket) do
+    {:ok,
+     socket
+     |> assign(name: name)}
+  end
+
+  @impl true
+  def render(%{platform_id: :swiftui} = assigns) do
+    ~Z"""
+    <Text modifiers={@native |> foreground_style(primary: {:color, :mint})}>
+      Hello <%= @name %>
+    </Text>
+    """swiftui
+  end
+
+  def render(assigns) do
+    ~H"""
+    <div>Hello <%= @name %></div>
+    """
+  end
+end
+|> KinoLiveViewNative.register("/hello/:name", ":index")
+```
+
+Go to http://localhost:5001/hello/josh
```

Comment:
This is why I'd like to avoid. This implementation requires the simulator to be restarted because the endpoint needs to be updated. That is very slow. Instead I'd like to stick to the single route and find a way to replace the active code for each section

Reply:
What do you mean? I'm not sure I understand the problem.

From the [tutorial](https://liveview-native.github.io/liveview-client-swiftui/tutorials/liveviewnative/03-navigation-and-hero), I imagine we would want users to register two routes:
```elixir
  scope "/", LvnTutorialWeb do
    pipe_through :browser

    live_session :cats do
      live "/cats", CatsListLive
      live "/cats/:name", CatLive
    end
  end
  ```

Reply:
Or is it that when we recompile the router, the native client on the simulator trips up and we need to restart it?

Reply:
@NduatiK oh ok, I see where you're coming from. We're not trying to reproduce the ios cats tutorial. I haven't even begun to sketch out the content for this tutorial yet but this is meant to introduce people to the concepts of demonstrate many of the SwiftUI views and modifiers. Navigation can probably be part of this too but what the immediate lesson will be is a single LV endpoint and template that can be updated to show off different SwiftUI view components.

Reply:
We can reduce functionality if it makes sense, I'm just not clear on what you would like to avoid.

> This is why I'd like to avoid.
I'm not too sure on what this means. If you are hoping to redefine the same LiveView in different cells that might not be possible in Elixir.

One solution would be to define `Server.HomeLive` in one cell then rename it to `Server.Home2Live` in the next cell. 
We could automate the renaming using macros so that users _think_ they are redefining the module but we are renaming it under the hood.

Reply:
@bcardarella, any feedback on this?

Reply:
My bad, I'll take a look today

Reply:
Just a note on redefining modules from the Dockyard Academy, looks like redefining modules is known "no-no":
https://github.com/DockYard-Academy/curriculum/issues/622

Reply:
I'm not super concerned about runtime perf issues related to module replacement. We could even make use of hot code loading if there is no public API. Remember, the Livebook is really only intended to be for a single person to play around with LiveView Native. Theoretically we could have a few people attach at the same time but we're that would be incredibly rare.

Reply:
Got this working by:
1. overriding `defmodule` inside our SmartCell
2. using a private Livebook API function (`Livebook.Runtime.Evaluator.delete_module(module_name)`) to delete modules that have already been registered.

---

I've also added moved configuration into a function.
Users just call `KinoLiveViewNative.start(app_name: "LiveView Native", port: 5001)` at the top of their notebook. See the example notebook.

Reply:
Nice :)

Reply:
Awesome, I'll merge in the changes.